### PR TITLE
convert mocha-avoid-only rule to use walk function

### LINKED ISSUE #680: Convert all existing rules to use a `walk` function instead of extending AbstractRule
Partner to #586: following palantir/tslint#2522, we should be recommending a ts.forEachChild strategy.

Accepting any PRs that take an existing rule and convert it to `walk`! ✨

### PR DESCRIPTION
**Pull request by drexler**
#### PR checklist

-   [x] Addresses an existing issue: fixes #680 

#### Overview of change:
Updates the mocha-avoid-only rule to use the walk function.

### CODE REVIEW

File: src/mochaAvoidOnlyRule.ts [hunk 1]
```diff
@@ -25,38 +26,46 @@ export class Rule extends Lint.Rules.AbstractRule {
...
-                            this.addFailureAt(node.getStart(), node.expression.getText().length, Rule.FAILURE_STRING_SPECIFY);
-                        } else if (node.expression.getText() === 'describe.only') {
-                            this.addFailureAt(node.getStart(), node.expression.getText().length, Rule.FAILURE_STRING_DESCRIBE);
-                        } else if (node.expression.getText() === 'context.only') {
-                            this.addFailureAt(node.getStart(), node.expression.getText().length, Rule.FAILURE_STRING_CONTEXT);
+                                case 'context.only':
+                                    ctx.addFailureAt(node.getStart(), text.length, Rule.FAILURE_STRING_CONTEXT);
+                                    break;
+
+                                default: // required per tslint rules for switch statements.
```
Comment (drexler) [inline src/mochaAvoidOnlyRule.ts]: Intentionally went with the `switch` statement over the `if/elseif` for clarity. The forced inclusion of `default` mars it, admittedly. I can switch to the `if/else if` if required.

**Comment by drexler** [comment]
build failure due to #684 😞

File: src/mochaAvoidOnlyRule.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/mochaAvoidOnlyRule.ts]: +1 for getting rid of continuous `.getText()` calls!

**Comment by JoshuaKGoldberg** [comment]
Build failures are pre-existing, unrelated and only on Node 6. #684

**Pull request merged**